src/frequenz/client/reporting/_client.py

```diff
@@ -157,6 +157,44 @@ async def list_single_component_data(
             for entry in page:
                 yield entry
 
+    # pylint: disable=too-many-arguments
+    async def list_multiple_microgrid_components_data(
+        self,
+        *,
+        microgrid_ids: list[int],
+        component_ids: list[int],
+        metrics: Metric | list[Metric],
+        start_dt: datetime,
+        end_dt: datetime,
+        page_size: int = 1000,
+    ) -> AsyncIterator[MetricSample]:
+        """Iterate over the data for multiple microgrids and components.
+
+        Args:
+            microgrid_ids: List of microgrid IDs.
+            component_ids: List of component IDs corresponding to each microgrid.
+            metrics: The metric name or list of metric names.
+            start_dt: The start date and time.
+            end_dt: The end date and time.
+            page_size: The page size.
+
+        Yields:
+            A named tuple with the following fields:
+            * timestamp: The timestamp of the metric sample.
+            * value: The metric value.
+        """
+        async for page in self._list_microgrid_components_data_pages(
```

Comment:
Since we should be as close as possible to the original function, we should use the same arguments as for `_list_microgrid_components_data_pages`

Reply:
Meaning that I should use `microgrid_components: list[tuple[int, list[int]]]` instead of `microgrid_ids: list[int], component_ids: list[int]`?

Reply:
Yes, because that reflects the data structure used in the underlying gRPC function.